app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@liyakun I thought it would be easy to get Dropzone working given the [simple tutorial](http://www.dropzonejs.com/#usage) and the existing code base, but if it's proving too difficult we can stick with the Rails upload field. The file size limit may already be restricted by the Web server, so we may just rely on that anyway.